be/src/pipeline/pipeline_fragment_context.cpp

```diff
@@ -993,4 +995,50 @@ std::string PipelineFragmentContext::debug_string() {
     return fmt::to_string(debug_string_buffer);
 }
 
+std::vector<std::shared_ptr<TRuntimeProfileTree>> PipelineFragmentContext::collect_profile_x()
+        const {
+    std::vector<std::shared_ptr<TRuntimeProfileTree>> res;
+    DCHECK(_query_ctx->enable_pipeline_x_exec() == true)
+            << fmt::format("Query {} calling a pipeline X function, but its pipeline X is disabled",
+                           print_id(this->_query_id));
+    std::stringstream ss;
+
+    for (auto& pipeline_profile : _runtime_state->pipeline_id_to_profile()) {
+        auto profile_ptr = std::make_shared<TRuntimeProfileTree>();
+        pipeline_profile->to_thrift(&(*profile_ptr));
+        res.push_back(profile_ptr);
+
+        std::vector<RuntimeProfile*> task_x_profile;
+        pipeline_profile->get_all_children(&task_x_profile);
+        for (RuntimeProfile* p : task_x_profile) {
+            if (p->name().find("PipelineXTask") != std::string::npos) {
+                ss << p->name() << '\n';
+            } else {
+                ss << '\t' << p->name() << '\n';
+            }
+        }
+    }
+
+    LOG_INFO("Query X {} fragment {} profile\n{} ", print_id(this->_query_id), this->_fragment_id,
+             ss.str());
+    return res;
+}
+
+std::shared_ptr<TRuntimeProfileTree> PipelineFragmentContext::collect_profile_p() const {
+    auto res = std::make_shared<TRuntimeProfileTree>();
+
+    DCHECK(_query_ctx->enable_pipeline_x_exec() == false) << fmt::format(
+            "Query {} calling a pipeline function, but it id using pipeline X, there are some "
+            "bugs.",
+            print_id(this->_query_id));
+
+    // std::stringstream ss;
+    DCHECK(_runtime_state->runtime_profile() != nullptr);
+    _runtime_state->runtime_profile()->compute_time_in_profile();
+    // _runtime_state->runtime_profile()->pretty_print(&ss);
+    _runtime_state->runtime_profile()->to_thrift(&(*res));
+
+    return res;
+}
```

Comment:
Should also add method in plan fragment executor

Reply:
> Should also add method in plan fragment executor

done
https://github.com/yiguolei/incubator-doris/pull/4/files#diff-03276016fa04b08d068c8eab318f3529bf0d153c0e23e7e006d4f790f2506efe